Add pg_stat_statements section (#952)
timescale/docs: cloud/metrics.md

# Metrics
You can view your service metrics from Timescale Cloud's [metrics
dashboard](#metrics-dashboard). This dashboard gives you service-level
information, such as CPU, memory, and storage usage.

> You can also view your query statistics by using the pre-installed

Reviewer: ```suggestion
You can also view your query-level statistics by using the pre-installed
```

[`pg_stat_statements`](#query-level-statistics-with-pg-stat-statements)
extension from a PostgreSQL client.

## Metrics dashboard

> Timescale Cloud provides a Metrics dashboard for managing your services. You can

Reviewer: ```suggestion
Timescale Cloud provides a metrics dashboard for managing your services. You can
```

> see the Metrics dashboard in your Timescale Cloud account by navigating to the

Reviewer: ```suggestion
see the metrics dashboard in your Timescale Cloud account by navigating to the
```

This indicates that metrics have not been collected for the period shown. It
does not mean that your Timescale Cloud service was down.

# Query-level statistics with pg_stat_statements
The `pg_stat_statements` extension gives you query-level statistics for your SQL
statements. It comes pre-installed with Timescale Cloud.

<highlight type="note">
For more information about `pg_stat_statements`, see the 
[PostgreSQL documentation](https://www.postgresql.org/docs/current/pgstatstatements.html).
</highlight>

> <highlight type="note">

Reviewer 2: ```suggestion
<highlight type="important">
```

You can view statistics for your queries through the `pg_stat_statements`

> extension, which provides a `pg_stat_statements` view. The recorded statistics

Reviewer: ```suggestion
extension. The recorded statistics
```

include the time spent planning and executing each query; the number of blocks
hit, read, and written; and more.

> You can query the `pg_stat_statements` view as you would any PostgreSQL view.

Reviewer: ```suggestion
You can query the `pg_stat_statements` view as you would in any PostgreSQL view.
```

The full view includes superuser queries, which are used by Timescale Cloud to

> manage your service in the background. To limit the results to only your

Reviewer: ```suggestion
manage your service in the background. To view only your
```

queries, filter by the current user.

Connect to your database using a PostgreSQL client, such as [`psql`][psql], and
run:
```sql
SELECT * FROM pg_stat_statements WHERE pg_get_userbyid(userid) = current_user;
```

> <highlight type="important">

Reviewer: May be this can go in the begining para of ## Query the pg_stat_statements view.

You cannot currently enable `track_io_timing` for your database. Statistics that
depend on `track_io_timing`, such as `blk_read_time` and `blk_write_time`, are
not collected.
</highlight>

## Example usage

> With `pg_stat_statements`, you can view performance statistics to help you

Reviewer: ```suggestion
With `pg_stat_statements`, you can view performance statistics that help you
```

monitor and optimize your queries.

Here are some sample scenarios to try.

> ### Identify long-running queries

Reviewer: ```suggestion
### Identifying long-running queries
```

> Identify the 5 longest-running queries by their mean execution time:

Reviewer: ```suggestion
To identify the five longest-running queries by their mean execution time:
```

```sql
SELECT calls,
    mean_exec_time,
    query
FROM pg_stat_statements
WHERE pg_get_userbyid(userid) = current_user
ORDER BY mean_exec_time DESC
LIMIT 5;
```

> ### Identify queries with highly variable execution time

Reviewer: ```suggestion
### Identifying queries with highly variable execution time
```
